### PR TITLE
[JENKINS-29492] Fix some findbugs issues

### PR DESCRIPTION
**Pull request by christ66**
https://issues.jenkins-ci.org/browse/JENKINS-29492

Fixing several findbugs issues:

* Use `%n` instead of `\n` to handle platform-specific line separators.
* Use `StringBuilder` for concatenating strings.
* Unnecessary unboxing of longs.
* Ignore the debugging static variable which can be mutable.

@reviewbybees

### CODE REVIEW

**Comment by recena** [comment]
@christ66 Please, add a ticket on Jira and modify the PR title.

**Comment by oleg-nenashev** [comment]
@recena 
The PR improves the code quality, but does not resolve real functional bugs. JIRA issues and even noting in the changelog are not required in such case.

:bee:

**Comment by recena** [comment]
@oleg-nenashev I totally disagree. Everything should be describe on a Jira ticket. Where are these rules?

**Comment by oleg-nenashev** [comment]
@recena 
There's no such rules, just a common practice in Jenkins project. Plugin maintainers can require such tickets, of course.

IMHO creating JIRA issues for refactorings/typo fixes does not help anyone:
* committers will spend more time on handling issues and changelogs
* users will spend more time on analyzing changelogs bloated by non-functional changes
* JIRA searches and monitoring will become more difficult

**Comment by recena** [comment]
@oleg-nenashev Thanks for your feefback

**Comment by christ66** [comment]
I created [JENKINS-29492](https://issues.jenkins-ci.org/browse/JENKINS-29492).

**Comment by recena** [comment]
@christ66 Thanks so much for add the ticket. It will be very easy to update the changelog.

**Comment by recena** [comment]
@christ66 Do you think this PR is ready to be merged?

**Comment by recena** [comment]
@christ66 Is it ready?

File: src/main/java/hudson/scm/SubversionChangeLogBuilder.java [hunk 1]
```diff
@@ -227,6 +227,8 @@ private static TransformerHandler createTransformerHandler() {
 
     private static final LocatorImpl DUMMY_LOCATOR = new LocatorImpl();
 
+    @edu.umd.cs.findbugs.annotations.SuppressWarnings(value = "MS_SHOULD_BE_FINAL",
```
Comment (jglick) [inline src/main/java/hudson/scm/SubversionChangeLogBuilder.java]: Use `@SuppressFBWarnings` and you do not need an FQN.

**Comment by jglick** [comment]
:bee:

**Pull request merged**